Mathlib/Algebra/Order/Group/Basic.lean

```diff
@@ -137,3 +137,72 @@ lemma zpow_eq_zpow_iff' (hn : n ≠ 0) : a ^ n = b ^ n ↔ a = b := zpow_left_in
 #align zsmul_eq_zsmul_iff' zsmul_eq_zsmul_iff'
 
 end LinearOrderedCommGroup
+
+section LinearOrderedAddCommGroupWithTop
+
+variable {α : Type*} [LinearOrderedAddCommGroupWithTop α]
+
+@[simp]
+lemma top_ne_zero' :
+    (⊤ : α) ≠ 0 := by
+  intro nh
+  have ⟨a, b, h⟩ := Nontrivial.exists_pair_ne (α := α)
+  have : a + 0 ≠ b + 0 := by simpa
+  rw [← nh] at this
+  simp at this
+
+@[simp] lemma neg_eq_top {a : α} : -a = ⊤ ↔ a = ⊤ where
+  mp h := by
+    by_contra nh
+    replace nh := add_neg_cancel_of_ne_top nh
+    rw [h, add_top] at nh
+    exact top_ne_zero' nh
+  mpr h := h ▸ neg_top
+
+@[simp]
+lemma add_eq_top_iff {a b : α} : a + b = ⊤ ↔ a = ⊤ ∨ b = ⊤ where
```

Comment:
Why the `_iff`? It's pretty unambiguous that the lemma is an iff. Compare with `mul_eq_zero`

Reply:
It collides with `ENNReal.add_eq_top`.

Reply:
Perhaps there should be some typeclass which `Mathlib.Algebra.Order.Monoid.WithTop` could use, which both `WithTop` and `LinearOrderedAddCommGroupWithTop` have instances of, as I see this is already a theorem there.

Reply:
Yes, that sounds right

Reply:
There is a problem that the corresponding theorems on `WithTop` have less requirements. For example `WithTop.add_eq_top` only requires `[Add α]`, while the `LinearOrderedAddCommMonoidWithTop` instance requires `[LinearOrderedAddCommMonoid α]`. How can this be resolved?

Reply:
You can keep `WithTop.add_eq_top`, but make it `protected`